eclipsesource/theia-cloud
# Add monitor theia extension

* Add monitor theia-extension
Added `monitor-theia` extension.
Works the same way as the monitor vscode extension.
Adjusted the way ports are replaced in deployment/service.
This allows to specify the same port for the monitor and the app itself.
Also make sure that the monitor not being enabled does not crash the operator.

* Delete theia-activity-tracker

* Adjust monitor values in value file

Contributed on behalf of STMicroelectronics.

Corresponding helm chart change: https://github.com/eclipsesource/theia-cloud-helm/pull/20

**To test:**
1. Build the operator and service images as `theia-cloud-(operator|service):local`
2. Pull the `simongraband/theia-cloud-demo:local` image. (It contains a plain theia application with the theia extension)
3. Retag the image to `theia-cloud-demo:local`
4. Checkout the helm chart PR
5. If you want you can adjust the monitor values in `valuesMinikubeMonitor.yaml` (e.g. reducing the `notifyAfter` and `timeoutAfter` values)
6. Do `helm install` using the checked out helm chart and the `valuesMinikubeMonitor.yaml` values file
7. See that the activity is tracked in the operator logs and that after your time limits a warning and a pod closed message are displayed to the user.
